.changeset/large-balloons-compete.md

```diff
@@ -0,0 +1,29 @@
+---
+'@astrojs/starlight': minor
+---
+
+Exposes the built-in localization system in the Starlight plugin `setup` hook.
+
+⚠️ **BREAKING CHANGE:**
+
+This addition changes how Starlight plugins add or update translation strings used in Starlight’s localization APIs.
+Plugins previously using the [`injectTranslations()`](https://starlight.astro.build/reference/plugins/#injecttranslations) callback function from the plugin [`setup`](https://starlight.astro.build/reference/plugins/#hookssetup) hook should now use the same function available in the [`init`](https://starlight.astro.build/reference/plugins/#hooksinit) hook.
+
+```diff
+export default {
+  name: 'plugin-with-translations',
+  hooks: {
+-   setup({ injectTranslations }) {
++   init({ injectTranslations }) {
```

Comment:
> Regarding the hook naming, I was wondering if we should follow the colon-separated pattern used by Astro integrations?

I personally like that idea.

> but maybe a bit misleading if you don’t set up i18n config here, only translations.

I guess, altho if we compare with our configuration file, we use the `locales` key to configure i18n, not `i18n` so I think it's fine.

> I guess we might have a config:done hook like Astro does that gives access to the final resolved Starlight config?

Maybe, I don't really see the need for it right now and I'm not even sure what you would do with it.

> which your current docs kind of suggests: “initialization function called before Starlight is initialized (during the astro:config:setup integration hook)”

Well, to be fair, like I said, I didn't really put any thought at all about the naming of the hook, and I documented it mostly to give an overview of all the changes and mimick what we had before.

I think if we ever need something else to be done before `config:setup` that is not `i18n`, we could still add an `xyz:setup` and call it before `config:setup` like we do with `i18n:setup` and it would be fine.

> If we do rename setup() it doesn’t have to be in this PR and can be done in a backwards compatible way where we still call setup() if defined but mark it as deprecated.

I think we maybe do want to do that at the same time, would be easier to communicate and organize docs? And we can still soft deprecate the `setup` method in the code so it's still run.

Reply:
OK, that sounds good! So if you think `i18n` is not problematic, I think it’s nice to have that concise hook name. And +1 to do the backwards compatible deprecation of `setup` in favour of `config:setup`.

Regarding what you might do with a `config:done`: most often I see the `astro:config:done` hook to share the final config with some other code via the integration closure, e.g.

```js
function integration() {
  let lazy = {};
  return {
    name: 'example',
    hooks: {
      'astro:config:setup'() {
        // add something referencing `lazy` e.g. a Vite plugin
        // that will run _after_ the done hook but wants the
        // final config shape
      },
      'astro:config:done'({config}) {
        lazy.config = config;
      },
    },
  };
}
```

But not suggesting we add that now. No point until some more clear needs emerge.

Reply:
Perfect, I'll get started asap on the new changes, thanks again.